src/annotations/_AnnotationManager.js

```diff
@@ -0,0 +1,224 @@
+import dcmjs from "dcmjs";
+
+import _MarkupManager from "./markups/_MarkupManager";
+import MeasurementMarkup from "./markups/measurement";
+import ArrowMarker from "./markers/arrow";
+import FreeTextMarkup from "./markups/freeText";
+import Enums from "../enums";
+import { getUnitsSuffix } from "./markups/utils";
+
+const { Marker, Markup, FeatureEvents } = Enums;
+
+class _AnnotationManager {
+  constructor({ map, source, controls, getROI } = {}) {
+    this.props = { map, source, controls, getROI };
+
+    /** Markups */
+    this[Markup.Measurement] = MeasurementMarkup(this.props);
+    this[Markup.FreeTextEvaluation] = FreeTextMarkup(this.props);
+
+    /** Markers */
+    this[Marker.Arrow] = ArrowMarker(this.props);
+
+    this.props.markupManager = new _MarkupManager({
```

Comment:
I have trouble understanding what's going on here. It's confusing that `this.props` gets passed to the constructor of the markup and marker classes without a `markupManager` property, but these classes internally access the `markupManager` property on the passed options object (`api`). See also above [comment](https://github.com/MGHComputationalPathology/dicom-microscopy-viewer/pull/43/files#r584433586).

Can this be refactored in a way that is more intuitive?

Reply:
Updated this logic, thanks